### ISSUE TITLE
Logging System Enhancement

### ISSUE THREAD

**Issue by qfrank** [status-im/status-react#21501]
## Background
Currently, our logging system has two inconsistencies:
1. Logs are disabled by default in release builds, making it difficult to diagnose user-reported issues
2. There's a mismatch between log file storage and configuration:
   - Log files: All users share the same log file
   - Log settings: Each user has their own logging configuration in the UI

## Problem Statement
These limitations affect our ability to effectively troubleshoot issues:
- When users experience login failures, we cannot collect error logs due to logging being disabled in release builds. See related [issue](https://github.com/status-im/status-mobile/issues/20996#issuecomment-2346722063).
- The current architecture, where multiple users share the same log file but have individual log configurations, creates confusion and potential conflicts

## Proposed Changes
1. Logging Configuration
   - Change default log level from DISABLED to ERROR in release builds
   - Implement global log level configuration instead of per-user settings to match the shared log file architecture

## Acceptance Criteria
1. Release builds should:
   - Have ERROR as the default log level
2. Log level settings should:
   - Be applied globally across the application
   - Be consistent with the shared log file approach
   - Remove the per-user log configuration from UI
   - User can set log level before login

cc @ilmotta @churik @Parveshdhull

**Comment by vkjr** [comment]
@qfrank, I'm totally support this idea.
One question regarding logs - wouldn't that be more convenient if we have different log files per user? I think they will be smaller and easier to investigate. 
I suppose the problem here is that we have some logs that precedes user login.. but maybe this can be a separate log file. 
Wdyt?

**Comment by qfrank** [comment]
> I suppose the problem here is that we have some logs that precedes user login.. but maybe this can be a separate log file.

I don't have a strong opinion on this. Either solution is okay. Thank you for your input! Let's see what else others think. I'll update the description when we have a conclusion :)

**Comment by churik** [comment]
Yes, I'm totally support this idea, besides seems desktop does not have this policy about logs, imo it is important to have the same policy across the app.

Also confirming @vkjr - for me it would make sense for each user to have their own log, but ofc it depends  on the cost of the development vs benefite, so I'd strat from simplest solution

**Comment by ilmotta** [comment]
### Problem 1

From the problems listed in this issue, to me the one that deserves the most attention and more immediately is `1. Logs are disabled by default in release builds, making it difficult to diagnose user-reported issues`.

> Change default log level from DISABLED to ERROR in release builds

Agreed 👍🏼 

I would like to bring another perspective about problem 1 because onboarding is a special flow where:

- There shouldn't be any private data to leak because the user hasn't logged in with any profile. It's also much easier for us to double-check the pre-login code and make sure it's safe regarding privacy (it's not a whole lot of code to check before the session starts, or is it?).
- The app is quiet (or should be) because there are almost no signals, no active profile, etc which means the pre-login logs won't (shouldn't) grow or grow so slowly that's insignificant.

Based on these points, I think it would be better to have a separate log file `prelogin.log` up until the point where a session actually starts, at which point status-go would switch and start to write to another file. This `prelogin.log` file would be global, unrelated to any profile and the log level could be `INFO` or even `DEBUG`. Its log level wouldn't be associated with the per-profile settings. We would want the maximum amount of information in these logs because they won't grow fast and they won't contain private data (guaranteed by us devs as much as we can), therefore, we can be more permissive instead of just the log level `ERROR` or `WARN`.

### Problem 2

> There's a mismatch between log file storage and configuration

Based on a quick discussion in the Core teams sync call this Monday Oct 28, people (including myself) tend to prefer to separate log files per profile. I think we should respect as much as possible the separation between profiles and logs are part of that. Another suggestion from @jrainville which I also like is to split log files per session.

But I think this Problem 2 is relatively low in our priorities @qfrank. Problem 1 would be very good to solve until 2.32 is published. I would say Problem 1 is kind of critical for users because some are getting locked out and we can't do much now to help.

**Comment by qfrank** [comment]
For problem 1, I would prefer solve it by simply changing default log level from DISABLED to ERROR in release builds as a quick fix. We could even include the quick fix in 2.31 if you think it's really critical for users, and then we can improve it like separating log file `prelogin.log` etc later? 🤔 @ilmotta

**Comment by ilmotta** [comment]
> For problem 1, I would prefer solve it by simply changing default log level from DISABLED to ERROR in release builds as a quick fix.

Fully agree @qfrank, the idea/solution with `prelogin.log` I shared wouldn't be for 2.31, maybe not even 2.32. And we haven't reached consensus on what's best yet.

> We could even include the quick fix in 2.31

I think that'd be great. Should be a simple and safe improvement and easy to verify if it works. We can still cherry-pick high-impact work. cc @churik

**Comment by ilmotta** [comment]
- Linking to https://github.com/status-im/status-mobile/pull/21864, as an example another problem related to pre-login logs being disabled and how enabling them with the current solution in `develop` is always problematic in way or another.

**Comment by churik** [comment]
@ilmotta 
should we pause working on this one for 2.34?

**Comment by ilmotta** [comment]
> [@ilmotta](https://github.com/ilmotta) should we pause working on this one for 2.34?

Good question 👍🏼 We never really stopped working on logging improvements. We are gradually making progress. The pre-login solution recently implemented by @qfrank came out of this GH issue, for example.

The following problem described above still exists to some extent:

> When users experience login failures, we cannot collect error logs due to logging being disabled in release builds. See related https://github.com/status-im/status-mobile/issues/20996#issuecomment-2346722063.

Frank did again great work on reducing the surface of leaking private data in the pre-login phase. Still, we only enabled logs at the error level.

Frank and I discussed about implementing a feature to allow the user to define the log level before login. Because the onboarding has more than one entrypoint, it's not as simple as adding a button and it could introduce a complexity to the onboarding, something we don't want. I'm thinking the shake feature or another gesture could be added to the app, so that it's not intrusive since it's quite an advanced feature (most users never even heard the concept of a "log level").

I'm going to create an issue for supporting switching the log level during onboarding. But I think this issue @qfrank we can close and continue to work on separate issues resolving individual problems. If you prefer @qfrank you can also move this issue to GH discussions since it feels more like a discussion.

**Comment by qfrank** [comment]
> > [@ilmotta](https://github.com/ilmotta) should we pause working on this one for 2.34?
> 
> Good question 👍🏼 We never really stopped working on logging improvements. We are gradually making progress. The pre-login solution recently implemented by [@qfrank](https://github.com/qfrank) came out of this GH issue, for example.
> 
> The following problem described above still exists to some extent:
> 
> > When users experience login failures, we cannot collect error logs due to logging being disabled in release builds. See related [#20996 (comment)](https://github.com/status-im/status-mobile/issues/20996#issuecomment-2346722063).
> 
> Frank did again great work on reducing the surface of leaking private data in the pre-login phase. Still, we only enabled logs at the error level.
> 
> Frank and I discussed about implementing a feature to allow the user to define the log level before login. Because the onboarding has more than one entrypoint, it's not as simple as adding a button and it could introduce a complexity to the onboarding, something we don't want. I'm thinking the shake feature or another gesture could be added to the app, so that it's not intrusive since it's quite an advanced feature (most users never even heard the concept of a "log level").
> 
> I'm going to create an issue for supporting switching the log level during onboarding. But I think this issue [@qfrank](https://github.com/qfrank) we can close and continue to work on separate issues resolving individual problems. If you prefer [@qfrank](https://github.com/qfrank) you can also move this issue to GH discussions since it feels more like a discussion.

let's close this once separate issues has been created

**Comment by churik** [comment]
I’ll exclude it from **2.34** since it’s ongoing and not directly tied to a specific milestone.  

Just a heads-up: the work you're doing on logs is truly amazing! The test team really appreciates it—it makes everyone’s life a lot easier. 🚀
